src/tr1d1um/communication.go

```diff
@@ -50,8 +54,24 @@ func (tr1 *Tr1SendAndHandle) Send(ch *ConversionHandler, resp http.ResponseWrite
 	//todo: any more headers to be added here
 	requestToServer.Header.Set("Content-Type", wrp.JSON.ContentType())
 	requestToServer.Header.Set("Authorization", req.Header.Get("Authorization"))
-	respFromServer, err = tr1.timedClient.Do(requestToServer)
-	return
+
+	ctx, cancel := context.WithTimeout(context.Background(), tr1.timeout)
+	defer cancel()
+
+	requestWithContext := requestToServer.WithContext(ctx)
+	ready := make(chan bool)
+
+	go func() {
+		respFromServer, err = tr1.client.Do(requestWithContext)
+		ready <- true
+	}()
+
+	select {
+	case <-ready:
```

Comment:
don't forget that this case is triggered when the channel is closed as well as when there's data on the channel.  this case can stay the same no matter what approach you take.